src/lexer.coffee

```diff
@@ -662,6 +662,7 @@ exports.Lexer = class Lexer
       @exportSpecifierList = no
 
     if value is ';'
+      @error 'unexpected ;' if prev?[0] in ['=', UNFINISHED...]
```

Comment:
do we not want `=` in `UNFINISHED`?

Reply:
or maybe the other way around: what does adding `in UNFINISHED` fixes, given that `prev?[0] is '='` is the relevant fix here? Same in `suppressSemicolons`.

Reply:
See https://github.com/jashkenas/coffeescript/issues/3098#issuecomment-26680094

`UNFINISHED` covers all the similar tokens like `+=` etc. I think @jashkenas in his comment was worried about the special cases around `=`, like `=(newline)(object literal)`.

Reply:
Also, putting `=` into `UNFINISHED` causes tests to fail.

Reply:
Ok. LGTM then.